### DISCUSSION TITLE
Debugging MediaCMS in VSCode

### DISCUSSION THREAD

**Discussion by Chocolate-Coco** [mediacms-io/mediacms#838] [Q&A]
Hello, has anyone managed to run the debugger in VSCode to debug MediaCMS?

When I try to start debugging, the virtual environment for Django starts successfully, but when I enter http://127.0.0.1:8000/ in the browser, I get an error 
```
TemplateDoesNotExist at /
cms/index.html

Request Method: 	GET
Request URL: 	http://127.0.0.1:8000/
Django Version: 	4.2.2
Exception Type: 	TemplateDoesNotExist
Exception Value: 	

cms/index.html

Exception Location: 	/home/mediacms.io/lib/python3.8/site-packages/django/template/loader.py, line 19, in get_template
Raised during: 	files.views.index
Python Executable: 	/home/mediacms.io/bin/python
Python Version: 	3.8.10
Python Path: 	

['/home/mediacms.io',
 '/home/mediacms.io/mediacms',
 '/home/ubuntu/.vscode/extensions/ms-python.python-2023.12.0/pythonFiles/lib/python/debugpy/_vendored/pydevd',
 '/usr/lib/python38.zip',
 '/usr/lib/python3.8',
 '/usr/lib/python3.8/lib-dynload',
 '/home/mediacms.io/lib/python3.8/site-packages']

Server time: 	Wed, 12 Jul 2023 13:02:02 +0100

```
So in settings.py I changed "DIRS": ["templates"] to "DIRS": [BASE_DIR + "/templates/"] it doesn't show the error anymore, but I have a clean white window....

Please, could someone help me???

**Comment by mgogoulos** [comment]
Hi, I have been working on this branch: https://github.com/mediacms-io/mediacms/tree/feat-dev-exp and very soon will open a PR, which improves the local development env (it starts Django in debug mode, so any change to python gets to be seen, and also React/frontend). I just need to make 1-2 more commits there and prepare a document, so you will be first to see/test it I hope!

**Reply by Chocolate-Coco** [reply]
That is great! I am unable to make improvements now, because I need to see, step by step (with debugger), how the application works. 
In return for your work I can share my improvements with you ;)

**Comment by mgogoulos** [comment]
I've just merged this PR https://github.com/mediacms-io/mediacms/pull/911 and will push a new latest release. 

Please check out the new page for local development and let me know how you find it (https://github.com/mediacms-io/mediacms/blob/main/docs/dev_exp.md)

I am going to write more but I believe it's a good start for now!
